drivers/net/ethernet/renesas/rswitch_tc_u32.c

```diff
@@ -183,7 +194,7 @@ static int rswitch_add_knode(struct net_device *ndev, struct tc_cls_u32_offload
 
 		if (is_tcf_act_skbmod(a)) {
 			/* skbmod dmac action can be offloaded only if placed before redirrect */
-			if (!rswitch_skbmod_can_offload(a) || filter.action != 0)
+			if (!rswitch_skbmod_can_offload(a) || (filter.action & ACTION_MIRRED_REDIRECT) != 0)
```

Comment:
Same check should be added for VLAN modify, it can not be made without redirect (due to L23 update).

Reply:
It is already there.